app/models/foreman_tasks/task.rb

```diff
@@ -76,6 +76,7 @@ def check_permissions_after_save
                                  :"foreman_tasks_links.resource_type" => resource.class.name)
            end)
     scope :for_action_types, (->(action_types) { where('foreman_tasks_tasks.label IN (?)', Array(action_types)) })
+    scope :duration, -> { select("DISTINCT foreman_tasks_tasks.*, coalesce(ended_at, current_timestamp) - coalesce(coalesce(started_at, ended_at), current_timestamp) as duration") }
```

Comment:
Since it returns a scope of tasks with calculated duration, maybe `with_duration` would be a better name?